### ISSUE TITLE
Top parament in plotting functions

### ISSUE THREAD

**Issue by Yumo-Xie** [chuiqin/irGSEA#37]
Hello,

Very helpful tool! Thank you for your work.

I've encounted an issue when using plotting functions from the package. The pathways showed in the plot seem to be chosen by their names rather than significance or degree of difference. Like in GOBP, I can always find pathways like GOBP_Acid and  GOBP_3UTR in the plot. Besides, if I specified top = 5, I only found pathways that begin with numbers following "GOBP". Is this how the parament designed to be, or maybe it is a bug?

Thank you!

**Comment by MorrisXie** [comment]
Well, I found it is related to the spread() function from tidyR. Because spread() do not keep the order of the original data https://github.com/tidyverse/tidyr/issues/453. 

For a quick solve, I changed the "Name" column into factor by adding ",Name = factor(Name, levels = unique(Name))" in the mutate() line 88, 107 and 184 of irGSEA.heatmap.R right before spread(). After these changes, everything seems to work fine.

The original function output, only pathways begining with "A" and numbers are considered as top.
<img width="423" alt="Screenshot 2024-03-14 at 14 46 53" src="https://github.com/chuiqin/irGSEA/assets/104150654/38b3c963-66e3-49f8-a32a-cded96e0301d">

After modification, the chosen pathways align with the integrated results.

<img width="585" alt="Screenshot 2024-03-14 at 14 47 09" src="https://github.com/chuiqin/irGSEA/assets/104150654/dc024624-0c1a-40c7-a168-afc92e1785e6">


I suppose similar changes can be applied to irGSEA.bubble function too. 

Hope this would help.
